src/main/java/com/redhat/red/build/koji/kerberos/KrbAuthenticator.java

```diff
@@ -182,129 +197,145 @@ else if ( ccache != null )
         return tgt;
     }
 
-    public static SgtTicket getSgt( KrbClient krbClient, String keytab, String ccache, String password, TgtTicket tgt, String serverPrincipal )
+    private static Credential findMatchingCredential( String principal, KrbCredentialCache credCache )
+            throws IOException
+    {
+        for ( Credential cred : credCache.getCredentials() )
+        {
+            if ( principal.equals( cred.getClientName().getName() ) )
+            {
+                return cred;
+            }
+        }
+
+        throw new IOException( "Could not find credential with client name matching principal " + principal
+                + " in credential cache" );
+    }
+
+    public static SgtTicket getSgt( KrbClient krbClient, String ccache, TgtTicket tgt, String serverPrincipal )
             throws KrbException
     {
-        KOptions requestOptions = new KOptions();
+        SgtTicket sgt;
 
-        if ( keytab != null )
+        if ( ccache != null )
         {
-            requestOptions.add( KrbOption.USE_KEYTAB, true );
-            requestOptions.add( KrbOption.KEYTAB_FILE, keytab );
+            File ccFile = new File( ccache );
+            sgt = krbClient.requestSgt( ccFile, serverPrincipal );
         }
         else
         {
-            requestOptions.add( KrbOption.USE_PASSWD, true );
-            requestOptions.add( KrbOption.USER_PASSWD, password );
+            sgt = krbClient.requestSgt( tgt, serverPrincipal );
         }
 
-        requestOptions.add( KrbOption.CLIENT_PRINCIPAL, tgt.getClientPrincipal().getName() );
-        requestOptions.add( KrbOption.SERVER_PRINCIPAL, serverPrincipal );
-        requestOptions.add( KrbOption.USE_TGT, tgt );
-
-        SgtTicket sgt = krbClient.requestSgt( requestOptions );
-
         return sgt;
     }
 
     public static ApReq makeReq( SgtTicket sgt )
             throws KrbException
     {
-        ApRequest apRequest = new ApRequest( sgt.getClientPrincipal(), sgt );
-
-        ApReq apReq = apRequest.getApReq();
+        ApRequest apRequest = new ApRequest( sgt.getClientPrincipal(), sgt, EnumSet.of( MUTUAL_REQUIRED ) );
```

Comment:
Does not seem to make a difference, but I think this option is used in the original koji code.